diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca593f2..7504161 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,10 +142,22 @@ if(WIN32)
 endif()
 
 if(MSVC)
+	if(MSVC_VERSION GREATER_EQUAL 1912)
+		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /Qspectre")
+	endif()
 	add_definitions(-Dinline=__inline)
 	message(STATUS "Using [${CMAKE_C_COMPILER_ID}] compiler")
 	if(CMAKE_C_COMPILER_ID MATCHES "MSVC" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
 		set(MSVC_DISABLED_WARNINGS_LIST
+			"C4242" # 'identifier' : conversion from 'type1' to 'type2', possible loss of data
+			"C4255" # 'function' : no function prototype given: converting '()' to '(void)'
+			"C4388" # 'expression': signed/unsigned mismatch
+			"C4464" # relative include path contains '..'
+			"C4668" # 'symbol' is not defined as a preprocessor macro, replacing with '0' for 'directives'
+			"C4710" # 'function' : function not inlined
+			"C4711" # function 'function' selected for inline expansion
+			"C4820" # 'bytes' bytes padding added after construct 'member_name'
+			"C5045" # Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
 			"C4018" # 'expression' : signed/unsigned mismatch
 			"C4057" # 'operator' : 'identifier1' indirection to
 		        	# slightly different base types from 'identifier2'
